mlflow/models/utils.py

```diff
@@ -695,6 +695,24 @@ def _is_scalar(x):
                     for value in pf_input.values()
                 ):
                     pf_input = pd.DataFrame([pf_input])
+                elif isinstance(pf_input, dict) and any(
+                    isinstance(value, np.ndarray) and value.ndim > 1 for value in pf_input.values()
+                ):
+                    # Pandas DataFrames can't be constructed with embedded multi-dimensional
```

Comment:
Looks like the same issue applies to nested list [[1,2,3]], should we add them as well?

Reply:
Great catch! Here's the current behavior for nested list on master:

Case 1: If a nested list is passed for a column defined in the signature, it is currently rejected with an "incompatible type" error.

Case 2: If a nested list is passed for a column that is not defined in the signature (extraneous column), it is silently dropped.

I've updated this PR to behave the same way for multi-dimensional arrays, since the "incompatible type" error in Case 1 is much clearer than filtering out the column, which would lead to an exception like "missing input for column X". This is confusing because column X was defined, it just had the wrong type.

Reply:
I've also added coverage for extraneous multi-dimensional lists

Reply:
>  If a nested list is passed for a column defined in the signature, it is currently rejected with an "incompatible type" error.

Do we plan to support multidim array in future ?

Reply:
Great question! IMO, we should go with something like JSON columns, which are even more flexible than array columns. Working on a proposal.